src/pages/index.js

```diff
@@ -103,101 +113,90 @@ const renCard = (data) => {
         );
         deleteCardPopup.open();
       },
+      handleLike: (card, isLiked) => {
+        cardElement.toggleLikeElement();
+        if (isLiked === true) {
+          api
+            .dislikeCard(card)
+            .then(() => {})
+            .catch((err) => {
+              console.error(err);
+            });
+        } else {
+          api
+            .likeCard(card)
+            .then(() => {})
+            .catch((err) => {
+              console.error(err);
+            });
+        }
+      },
     },
     cardSelector
   );
 
   cardSection.addItem(cardElement.getView());
 };
-// render card function
-
-const renderCard = (data) => {
-  const cardElement = new Card(
-    {
-      data,
-      handleImagePreview: (imageData) => {
-        previewImage.open(imageData);
-      },
-      handleDeleteIcon: (cardElement) => {
-        const deleteCardPopup = new PopupDelete(
-          deletePopupSelector,
-          deleteButtonSelector,
 
-          () => {
-            // console.log(cardElement.id);
+// popup with form - edit avatar
 
-            api
-              .deleteCard(cardElement)
-              .then(() => {
-                if (cardElement) {
-                  cardElement.remove();
-                }
-              })
-              .catch((err) => {
-                console.error(err);
-              });
-          }
-        );
-        deleteCardPopup.open();
-      },
-    },
-    cardSelector
-  );
-  cardSection.addItem(cardElement.getView());
-};
+const editAvatarPopUp = new PopupWithForm(formAvatarSelector, (info) => {
+  editAvatarPopUp.uploadingInfo(true);
+  api
+    .editUserAvatar(info)
+    .then((info) => {
+      setAvatar(info);
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editAvatarPopUp.uploadingInfo(false);
+      editAvatarPopUp.close();
```

Comment:
`editAvatarPopUp.close();`

* You need to close popups only in blocks `then` after a **successful response** from the server so that when there is an error it would be opened (the inputs will not be cleared and the user could send the data again to the server) and if you change the button text you could see the change as well